lib/components/sidebar-component.js

```diff
@@ -53,16 +53,18 @@ export default class SidebarComponent extends BaseContainer {
 	// Need to click header on mobile to focus the sidebar menu
 	ensureSidebarMenuVisible() {
 	}
+
 	searchForSite( searchString ) {
 		const searchSelector = By.css( '.site-selector input[type="search"]' );
 		const siteSelector = By.css( `.site-selector .site a[aria-label*="${searchString}"]` );
 
 		this.driver.findElement( searchSelector ).then( ( searchField ) => {
 			return searchField.isDisplayed().then( ( searchEnabled ) => {
 				if ( searchEnabled ) {
-					driverHelper.setWhenSettable( this.driver, searchSelector, searchString );
+					return driverHelper.setWhenSettable( this.driver, searchSelector, searchString ).then( () => {
+						return driverHelper.clickWhenClickable( this.driver, siteSelector );
```

Comment:
Is it necessary to chain this on the `then()` here? I would have thought it would work as two separate lines.

Reply:
Not actually, but when you chain all the async calls - you can be sure that everything works as expected, and commands are called one by one